src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright 2024 Mia srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ReactElement, useCallback, useMemo, useState } from 'react'
+import { Modal as AntModal } from 'antd'
+import classNames from 'classnames'
+
+import { Shape, Type } from '../Button/Button.types'
+import { BodyM } from '../Typography/BodyX/BodyM'
+import { Button } from '../Button'
+import { H4 } from '../Typography/HX/H4'
+import { Icon } from '../Icon'
+import { ModalProps } from './Modal.props'
+import { Size } from './Modal.types'
+import styles from './Modal.module.css'
+import { useTheme } from '../../hooks/useTheme'
+
+const { Circle } = Shape
+const { Ghost } = Type
+const { Small, Large, FullScreen } = Size
+const {
+  asideLabel,
+  asideLabelWrapper,
+  body,
+  bodyFullWidth,
+  content,
+  contentWrapper,
+  footerButtons,
+  modal,
+  modalSm,
+  modalLg,
+  modalFs,
+} = styles
+
+/**
+ * Modal dialog that opens in overlay and allows to perform flows of actions, or to show specific
+ * but less relevant information than those displayed on the underlying page.
+ *
+ * @link https://ant.design/components/modal
+ * @returns {Modal} Modal component
+ */
+export const Modal = ({
+  aside,
+  children,
+  docLink,
+  footer,
+  isBodyFullWidth,
+  isClosable,
+  isVisible,
+  onCloseClick,
+  size,
+  title,
+}: ModalProps): ReactElement => {
+  const { palette } = useTheme()
+
+  const docLinkIcon = useMemo(() => (
+    <Icon color={palette?.action?.link?.active} name="PiBookOpen" size={16} />
+  ), [palette?.action?.link?.active])
+
+  const onClickDocLink = useCallback(() => window.open(docLink, '_blank'), [docLink])
+
+  const modalClassNames = useMemo(() => classNames([
+    modal,
+    size === Small && modalSm,
+    size === Large && modalLg,
+    size === FullScreen && modalFs,
+  ]), [size])
+  const modalBodyClassNames = useMemo(() => classNames([
+    body,
+    isBodyFullWidth && bodyFullWidth,
+  ]), [isBodyFullWidth])
+  const modalContentWrapperClassNames = useMemo(() => classNames([
+    content,
+    size !== Small && aside && contentWrapper,
+  ]), [aside, size])
+
+  const [isAsideOpen, setIsAsideOpen] = useState(false)
+
+  const modalTitle = useMemo(() => (
```

Comment:
these memoized components could be full fledged components, this would make evolving them easier and this Modal codebase far more shorter